### PR TITLE
Attempt to convert frc castling to input planes.

### PR DESCRIPTION
**Pull request by Tilps**
My brain hurts...

This needs lots of testing...

### CODE REVIEW

**Comment by Tilps** [comment]
bitorder little introduces a dependency on numpy 1.17 - I might just rewrite it with a reversing slice.

**Comment by Tilps** [comment]
I don't think I have a good way to test this yet - definitely some risk I've got the byte order or bit orders reversed in some place.

File: tf/chunkparser.py [hunk 1]
```diff
@@ -207,20 +207,33 @@ def convert_v5_to_tuple(self, content):
...
         # detect edges of the board more easily
+        if input_format == 1:
+            castling_planes = self.flat_planes[us_ooo] + \
+                              self.flat_planes[us_oo] + \
+                              self.flat_planes[them_ooo] + \
+                              self.flat_planes[them_oo]
+        elif input_format == 2:
+            # Each inner array has to be reversed as these fields are in opposite endian to the planes data.
+            them_ooo_bytes = np.unpackbits(np.array([them_ooo], dtype=np.uint8))[:, ::-1].astype(np.float32).tobytes()
+            us_ooo_bytes = np.unpackbits(np.array([us_ooo], dtype=np.uint8))[:, ::-1].astype(np.float32).tobytes()
```
Comment (mooskagh) [inline tf/chunkparser.py]:
```bash
pip install yapf
yapf --style=pep8 chunkparser.py  # or --style=google
```
would split those long lines. :-P

Comment (Tilps) [reply tf/chunkparser.py]: Yes our training scripts could with a good formatting.

**Comment by Tilps** [comment]
Step 1 to testing it is actually having training data with format 2, except that isn't step 1 because that requires a network with input format 2, except that isn't step 1 because that requires training code to make such a net.

So... I'm going to submit it for now under the assumption that I have at least properly guarded the new code away from the default case - and address any details once I write another PR to create FRC nets in the first place.

**Pull request merged**